### ISSUE TITLE
file image mount `rw` option issue in v2.39 ?

### ISSUE THREAD

**Issue by suiryc** [util-linux/util-linux#2305]
Hi,

There seems to be an issue using the `rw` option when mounting file images in v2.39 on ArchLinux.
The issue occurs when giving `rw` as part of mounting options, whether passed using the `--options` CLI parameter of `mount` or when relying on `/etc/fstab` configuration. It also occurs when using the `--rw` CLI parameter.

Steps to reproduce:

1. Create and format an image file
```
dd if=/dev/zero of=/tmp/fs_10M.ext4 bs=1M count=10
mkfs.ext4 /tmp/fs_10M.ext4
```

2. Try to mount it, with the `rw` option
```
mount --options "rw,relatime,data=ordered" /tmp/fs_10M.ext4 /mnt
```

Result:
```
mount: /mnt: WARNING: source write-protected, mounted read-only.
```
And the mountpoint is indeed read-only.
Alternatively, using the `--rw` CLI parameter:
```
mount --rw --options "relatime,data=ordered" /tmp/fs_10M.ext4 /mnt
```

Result:
```
mount: /mnt/tmp: /dev/loop1 is write-protected but explicit read-write mode requested.
       dmesg(1) may have more information after failed mount system call.
```
`dmesg` do not appear to give useful details other than `write access unavailable, skipping orphan cleanup`.

Using `rw` alone or with other options does not seem to matter. The order of listed options also does not seem to matter.
Doubling it, e.g. `--options "rw,rw,relatime,data=ordered"` (again, order does not seem to matter) works.
Doubling `--rw` also works, as well as using both `--rw` and `rw` in `--options`.

Version 2.38 works.

Tested with v2.39-1, v2.39-4 and v2.39-8 from ArchLinux (rolling release).
Is it only related to the version built from ArchLinux or is the issue upstream ?

**Comment by t-8ch** [comment]
This is an issue upstream, too.
FYI if you do not specify "rw", "--rw" at all, it also works.

But indeed this is a bug.

**Comment by suiryc** [comment]
Fixed with #2306
